src/components/Consents/Consents.js

```diff
@@ -14,89 +18,37 @@ import {
 const regexHrefOpenTag = new RegExp(/<a(.|\n)*?>/);
 const regexHrefCloseTag = new RegExp(/<\/a(.|\n)*?>/);
 
-export class Consents extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      consentDefinitions: [],
-      checked: [],
-      consentsLabels: [],
-      consentLoaded: false,
-      generalError: ''
-    };
-  }
-
-  componentDidMount() {
-    const { publisherId } = this.props;
-    if (publisherId) {
-      this.getConsents(publisherId).then(() => {});
-    }
-  }
+const Consents = ({ error, onChangeFn, t }) => {
+  const {
+    publisherConsents,
+    checked,
+    loading,
+    error: generalError
+  } = useSelector(state => state.publisherConsents);
+  const { publisherId } = useSelector(state => state.publisherConfig);
 
-  componentDidUpdate(prevProps) {
-    const { publisherId } = this.props;
+  const dispatch = useDispatch();
 
-    if (prevProps.publisherId !== publisherId) {
-      this.getConsents(publisherId).then(() => {
-        this.validateConsents();
-      });
-    }
-  }
-
-  getConsents = async publisherId => {
-    try {
-      const consentsIncome = await getConsents(publisherId);
-      if (consentsIncome.responseData && consentsIncome.responseData.consents) {
-        const consentsDetails = consentsIncome.responseData.consents.map(
-          element => {
-            return {
-              name: element.name,
-              version: element.version,
-              required: element.required
-            };
-          }
-        );
-        const labels = consentsIncome.responseData.consents.map(
-          element => element.label
-        );
-        const initArray = new Array(consentsDetails.length).fill(false);
-        this.setState({
-          consentDefinitions: consentsDetails,
-          consentLoaded: true,
-          consentsLabels: labels,
-          checked: initArray
-        });
-      } else if (consentsIncome.errors.includes('Invalid param publisherId')) {
-        const { disabledRegisterButton } = this.props;
-        this.setState({
-          consentLoaded: true,
-          generalError: 'noPublisherId'
-        });
-        disabledRegisterButton();
-      }
-    } catch (error) {
-      return error;
-    }
-    return false;
+  const validateConsents = () => {
+    onChangeFn(checked, publisherConsents);
```

Comment:
to double-check the validation: 
![Screen Shot 2023-02-27 at 9 53 26 AM](https://user-images.githubusercontent.com/41182484/221517716-8ef82afd-b8fd-42e4-95f6-d528eb60767f.png)
When user clicks the Register button the required fields should be highlighted. Also, required consents should have '*' at the end
![Screen Shot 2023-02-27 at 9 55 39 AM](https://user-images.githubusercontent.com/41182484/221518063-9a9d6426-9982-4b2a-a6bf-7fe076ae9243.png)